src/Surfnet/StepupSelfService/SelfServiceBundle/Controller/RegistrationController.php

```diff
@@ -178,9 +179,25 @@ public function registrationPdfAction($secondFactorId)
                 'tempDir' => sys_get_temp_dir(),
             )
         );
+        $mpdf->setLogger($this->get('logger'));
+
         $mpdf->WriteHTML($content);
-        $mpdf->Output('registration-code.pdf', MpdfDestination::DOWNLOAD);
+        $output = $mpdf->Output('registration-code.pdf', MpdfDestination::STRING_RETURN);
+
+        $response = new Response($output);
+        $response->headers->makeDisposition(
+            ResponseHeaderBag::DISPOSITION_ATTACHMENT,
+            'registration-code.pdf'
+        );
+
+        $response->headers->set('Content-Description', 'File Transfer');
+        $response->headers->set('Content-Transfer-Encoding', 'binary');
+        $response->headers->set('Cache-Control', 'public, must-revalidate, max-age=0');
+        $response->headers->set('Pragma', 'public');
+        $response->headers->set('Expires', 'Sat, 26 Jul 1997 05:00:00 GMT');
+        $response->headers->set('Last-Modified', '' . gmdate('D, d M Y H:i:s') . ' GMT');
+        $response->headers->set('Content-Type', 'application/pdf');
 
-        exit;
+        return $response;
```

Comment:
The PDF is not presented to the user as a download but opens in the PDF viewer of the browser.

Reply:
I had to set the disposition header myself. I've added this so now it should work as expected.